### PR TITLE
#971 unique generator values

### LINKED ISSUE #971: Unique modifier for Generators
Filters out any value previously seen.

### PR DESCRIPTION
**Pull request by atmikev**
Fixes #971 

@sksamuel Not sure if this is a good idea or not, but wanted to throw it out there to see if I should keep refining it.

Also, it didn't look like it in the IDE but I think my indentation is off?

### CODE REVIEW

**Comment by atmikev** [comment]
Fixed indentation issues.

File: kotest-assertions/src/commonMain/kotlin/io/kotest/properties/Gen.kt [hunk 1]
```diff
@@ -46,6 +47,10 @@ interface Gen<T> {
 
   fun shrinker(): Shrinker<T>? = null
 
+  fun uniqueRandoms(seed: Long? = null): Sequence<T> = sequence {
+     yieldAll(random(seed).distinct())
+  }
```
Comment (sksamuel) [inline kotest-assertions/src/commonMain/kotlin/io/kotest/properties/Gen.kt]: Can we make this an extension method on Generator and not defined inside the interface itself. It's fine to keep it in the same file just below the interface definition.

**Review by sksamuel — CHANGES_REQUESTED** [review]

File: kotest-assertions/src/commonMain/kotlin/io/kotest/properties/Gen.kt (hunk 1 above)
Comment (atmikev) [reply kotest-assertions/src/commonMain/kotlin/io/kotest/properties/Gen.kt]: Updated

**Comment by sksamuel** [comment]
:+1:

**Pull request merged**